[DCP] Improve error handling for process based async checkpointing

meetv18 opened the pull request:
Summary:
### PR Context
- Kill background process only when PG init fails or there is an explicit `TERMINATE` signal from main process. 
- When a checkpoint fails to save, log and return the error but continue the serving loop.

Test Plan:
CI

Rollback Plan:

Differential Revision: D79177410




cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta

Skylion007 commented on torch/distributed/checkpoint/_async_process_executor.py:
```diff
@@ -224,22 +233,23 @@ def _checkpointing_subprocess(
+                    )
+                    parent_conn.send(response)
+                    logger.info(
+                        f"Completed checkpoint save request for checkpoint_id={obj.checkpoint_request_id}"  # noqa: G004
```
Please avoid unnecessary fstring formatting particulraly for info calls, use lazy % formatting

meetv18 committed:
[DCP] Improve error handling for process based async checkpointing (#159374)

Summary:

### PR Context
- Kill background process only when PG init fails or there is an explicit `TERMINATE` signal from main process. 
- When a checkpoint fails to save, log and return the error but continue the serving loop.

Test Plan:
CI

Rollback Plan:

Reviewed By: sibuachu

Differential Revision: D79177410
